extensions/hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3Repository.java

```diff
@@ -0,0 +1,211 @@
+            md5Hash16 = BaseEncoding.base16().lowerCase().encode(mdMD5.digest());
+        } catch (final IOException e) {
+            throw new ArtifactStoreException(e.getMessage(), e);
+        }
+
+        return store(sha1Hash16, md5Hash16, contentType, file, hash);
+    }
+
+    private DbArtifact store(final String sha1Hash16, final String mdMD5Hash16, final String contentType,
+            final File file, final DbArtifactHash hash) {
+        final S3Artifact s3Artifact = new S3Artifact(amazonS3, s3Properties, sha1Hash16);
+        checkHashes(s3Artifact, hash);
+        LOG.info("Storing file {} with length {} to AWS S3 bucket as SHA1 {}", file.getName(), file.length(),
+                s3Properties.getBucketName(), sha1Hash16);
+        try (final InputStream inputStream = new BufferedInputStream(new FileInputStream(file),
+                RequestClientOptions.DEFAULT_STREAM_BUFFER_SIZE)) {
+
+            s3Artifact.setContentType(contentType);
+            s3Artifact.setArtifactId(sha1Hash16);
+            s3Artifact.setSize(file.length());
+            s3Artifact.setContentType(contentType);
+            s3Artifact.setHashes(new DbArtifactHash(sha1Hash16, mdMD5Hash16));
+
+            final ObjectMetadata objectMetadata = new ObjectMetadata();
+            final String mdMD5Hash64 = BaseEncoding.base64()
+                    .encode(BaseEncoding.base16().lowerCase().decode(mdMD5Hash16));
+            objectMetadata.setContentMD5(mdMD5Hash64);
+            objectMetadata.setContentType(contentType);
+            objectMetadata.setContentLength(file.length());
+            objectMetadata.setHeader("x-amz-meta-md5chksum", mdMD5Hash64);
+            if (s3Properties.isServerSideEncryption()) {
+                objectMetadata.setHeader(Headers.SERVER_SIDE_ENCRYPTION,
+                        s3Properties.getServerSideEncryptionAlgorithm());
+            }
+
+            if (exists(sha1Hash16)) {
+                return s3Artifact;
+            }
+            amazonS3.putObject(s3Properties.getBucketName(), sha1Hash16, inputStream, objectMetadata);
+
+        } catch (final IOException | AmazonClientException e) {
+            throw new ArtifactStoreException(e.getMessage(), e);
+        } finally {
+            file.delete();
+        }
+        return s3Artifact;
+    }
+
+    @Override
+    public void deleteBySha1(final String sha1Hash) {
+        LOG.info("Deleting S3 object from bucket {} and key {}", s3Properties.getBucketName(), sha1Hash);
+        amazonS3.deleteObject(new DeleteObjectRequest(s3Properties.getBucketName(), sha1Hash));
+    }
+
+    @Override
+    public DbArtifact getArtifactBySha1(final String sha1) {
+        LOG.info("Retrieving S3 object from bucket {} and key {}", s3Properties.getBucketName(), sha1);
+        final GetObjectRequest getObjectRequest = new GetObjectRequest(s3Properties.getBucketName(), sha1);
+        final S3Object s3Object = amazonS3.getObject(getObjectRequest);
```

Comment:
I don't think we need the advanced options provided by `GetObjectRequest`, so better go for 
`amazonS3.getObject(String bucketName, String key)`.